packages/kbn-config/src/index.ts

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export {
+  applyDeprecations,
+  ConfigDeprecation,
+  ConfigDeprecationFactory,
+  configDeprecationFactory,
+  ConfigDeprecationLogger,
+  ConfigDeprecationProvider,
+  ConfigDeprecationWithContext,
+} from './deprecation';
+
+export { RawConfigurationProvider, RawConfigService } from './raw';
+
+export { ConfigService, IConfigService } from './config_service';
+export { Config, ConfigPath, isConfigPath, hasConfigPathIntersection } from './config';
+export { ObjectToConfigAdapter } from './object_to_config_adapter';
+export { CliArgs, Env } from './env';
+export { EnvironmentMode, PackageInfo } from './types';
+export { LegacyObjectToConfigAdapter } from './legacy';
+
+// mocks
+export { configMock } from './config.mock';
+export { configServiceMock } from './config_service.mock';
+export { rawConfigServiceMock } from './raw/raw_config_service.mock';
+export { getEnvOptions } from './__mocks__/env';
```

Comment:
As these mocks are used in `core`, I had to expose them on the package's entrypoint. This feels wrong, but I think creating an additional `@kbn/config-mocks` package is overkill, wdyt?

I couldn't find a way to have something like `import { rawConfigServiceMock } from '@kbn/config/mocks'`, as I was forced to do ``import { rawConfigServiceMock } from '@kbn/config/target/out/mocks'``. Maybe someone has more knowledge than me on npm packages?

Reply:
Our nodejs version doesn't support `exports` https://nodejs.org/docs/latest-v12.x/api/esm.html#esm_main_entry_point_export
We could emit `*.js` files next to `*.ts` and ignore them in `.gitignore`.

Reply:
Since we build in place and can't produce the JS at the root of the module, I don't see any other alternative. Sounds like the best thing right now would be to use target in the import until we upgrade to v12.